packages/odyssey-react-mui/src/Tile.tsx

```diff
@@ -0,0 +1,147 @@
+import { MoreIcon } from "./icons.generated";
+import { HtmlProps } from "./HtmlProps";
+import styled from "@emotion/styled";
+import {
+  DesignTokens,
+  useOdysseyDesignTokens,
+} from "./OdysseyDesignTokensContext";
+import { MenuButton } from "./MenuButton";
+
+export type TileProps = {
+  description?: string;
+  image?: ReactElement | NullElement; // Icon or image
+  menuItems?: ReactElement;
+  overline?: string;
+  title?: string;
+} & ( // You can't have actions and onClick at the same time
+  | {
+      onClick: () => void;
+      button?: never;
+      menuItems?: never;
+    }
+  | {
+      onClick?: never;
+      button?: ReactElement<typeof Button> | NullElement;
+      menuItems?: ReactElement;
+    }
+) &
+  HtmlProps;
+
+const ImageContainer = styled.div<{
+  odysseyDesignTokens: DesignTokens;
+  hasMenuItems: boolean;
+}>`
+  display: flex;
+  align-items: flex-start;
+  max-height: 64px;
+  margin-block-end: ${(props) => props.odysseyDesignTokens.Spacing5};
+  padding-right: ${(props) =>
+    props.hasMenuItems ? props.odysseyDesignTokens.Spacing5 : 0};
+`;
+
+const MenuButtonContainer = styled.div<{ odysseyDesignTokens: DesignTokens }>`
+  position: absolute;
+  right: ${(props) => props.odysseyDesignTokens.Spacing3};
+  top: ${(props) => props.odysseyDesignTokens.Spacing3};
+`;
+
+const Tile = ({
+  button,
+  description,
+  image,
+  menuItems,
+  onClick,
+  overline,
+  title,
+}: TileProps) => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+
+  const cardContent = useMemo(() => {
+    return (
+      <>
+        {image && (
+          <ImageContainer
+            odysseyDesignTokens={odysseyDesignTokens}
+            hasMenuItems={Boolean(menuItems)}
+          >
+            {image}
+          </ImageContainer>
+        )}
+
+        {overline && <Support component="div">{overline}</Support>}
+        {title && <Heading5 component="div">{title}</Heading5>}
+        {description && (
+          <Paragraph color="textSecondary">{description}</Paragraph>
+        )}
+
+        {button && (
+          <MuiCardActions>
+            <ButtonContext.Provider value={{ isFullWidth: true }}>
+              {button}
+            </ButtonContext.Provider>
+          </MuiCardActions>
+        )}
+      </>
+    );
+  }, [
+    button,
+    description,
+    image,
+    menuItems,
+    overline,
+    title,
+    odysseyDesignTokens,
+  ]);
+
+  return (
+    <MuiCard className={onClick ? "isClickable" : ""}>
+      {onClick && (
+        <MuiCardActionArea onClick={onClick}>{cardContent}</MuiCardActionArea>
+      )}
+
+      {!onClick && cardContent}
+
+      {menuItems && (
+        <MenuButtonContainer odysseyDesignTokens={odysseyDesignTokens}>
+          <MenuButton
+            endIcon={<MoreIcon />}
+            ariaLabel="Tile menu"
+            buttonVariant="floating"
+            menuAlignment="right"
+            size="small"
+          >
+            {menuItems}
+          </MenuButton>
```

Comment:
This says `menuItems`, but it's passed as `children` to `MenuButton`. Doesn't `MenuButton` only take text?

Reply:
`MenuButton` takes `children`. These are the items that are rendered in the actual drop down menu.

@jordankoschei-okta We probably want some kind of `tooltipText` here since this is an icon only button right? Speaking of, we should probably make sure that is required when `buttonLabel` is undefined right?